packages/jest-jasmine2/src/index.js

```diff
@@ -43,16 +43,31 @@ function jasmine2(config, environment, moduleLoader, testPath) {
       moduleLoader.requireModule(null, config.setupTestFrameworkScriptFile);
     }
   });
-
+  function isSpyLike(test) {
+    return !! (test.calls && test.calls.all);
+  }
+  function isMockLike(test) {
+    return test.mock !== undefined;
+  }
   env.beforeEach(() => {
     jasmine.addMatchers({
       toBeCalled: (/* util, customEqualityTesters */) => {
         return {
-          compare: (actual/*, expected */) => {
-            if (actual.mock === undefined) {
-              throw Error('toBeCalled() should be used on a mock function');
+          compare: (actual, expected) => {
+            if (expected) {
+              throw Error(
+                'toBeCalled() does not accept parameters, use toBeCalledWith instead'
+              );
+            }
+            if (!isMockLike(actual) && !isSpyLike(actual)) {
+              throw Error('toBeCalled() should be used on a mock function or a jasmine spy');
+            }
+            let calls = actual.mock.calls;
```

Comment:
yeah, I add the two function later and I miss that 
(also I should force cast to Boolean in the jasmine 1 implementation for consistency)